### PR TITLE
Community moderation offer

### PR DESCRIPTION
**Pull request by GreenAsJade**
Fixes people getting community moderator powers without being asked first :)

## Proposed Changes

  - People get a notification that they have been offered community moderator powers
  - They can accept or reject
  - Moderators can see where they are up to (accepted/pending/rejected)
  - The notification is dismissable only by accepting or rejecting the powers

Needs https://github.com/online-go/ogs/pull/1837

### CODE REVIEW

**Comment by GreenAsJade** [comment]
UI needs tweaking and placeholder text replaced.    Updates for that welcome!

RDH intro to the powers next.

File: src/components/Notifications/ModerationOfferModal.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                    <p>
+                        {pgettext(
+                            "Part of the description offering community moderation powers to a user.",
+                            "You'll get access to tools that will allow you to vote on reports raised by users - that will tell us how to handle those reports.",
+                        )}
+                    </p>
+                    <p>
+                        {pgettext(
+                            "Part of the description offering community moderation powers to a user.",
+                            "We just need your agreement to use the powers in the best interests of the OGS community, and in line with current policies and practices.",
```
Comment (anoek) [inline src/components/Notifications/ModerationOfferModal.tsx:104]: We should have a link to said policies and practices.. and which I guess means writing such things too :sweat_smile:

Comment (GreenAsJade) [reply src/components/Notifications/ModerationOfferModal.tsx:104]: That's why I wrote "practices" as well as "policies" 😝   The assumption being that anyone getting this kinda knows the deal even if it isn't written down formally 😅

**Comment by anoek** [comment]
Logic all lgtm. I think I'll want to wordsmith and gussy up the messaging a little bit still before we flip on the automatic stuff, make official policies and guidelines, stuff like that, but this is good enough for now for sure.

**Pull request merged**